plugins/README.md

```diff
@@ -1,41 +1,96 @@
 
+You can also have several HTML file if needed, but be aware that since your plugin will be contained in a `<iframe>`, your user won't have access to the navigation history and arrows.
+
+For an example of a simple plugin, you can look at [marcel-plugin-text](https://github.com/EmrysMyrddin/marcel-plugin-text).
+
+### Structure of the frontend / index.html
+
+The first thing you will need is [marcel.js](https://github.com/EmrysMyrddin/marcel-plugin-text/raw/master/frontend/marcel.js), which is a small script that will help you handle the interaction with MARCEL.
+Include it in your HTML:
 ```html
-<link rel="import" href="/plugins/shared-styles/shared-styles.html" />
+<script src="./marcel.js"></script>
 ```
 
-And then between the two template tags :
+Then, you should create a new class that inherit `Marcel.Plugin`:
+```javascript
+class MyPlugin extends Marcel.plugin {
+    <...>
+}
+```
+Your class should implement two function: 
 
-```html
- <style include="shared-styles"></style>
+1. The render function:
+
+```javascript
+render() {
+    const { props1, props2 } = this.props
+}
 ```
 
-As the styles, you can also import other plugins to use them. For example in the welcome plugin, if you want to use the datetime plugin, you have to import it:
+This function is called every time MARCEL need to render your plugin. 
+As the name of the function imply, you should do everything that involve rendering your page in this function.
+The object `this.props` will contain every properties sent by MARCEL.
 
-```html
-<link rel="import" href="/plugins/marcel-plugin-datetime" />
+2. The propsDidChange function:
+
+```javascript
+propsDidChange(prevProps) {
+    const { props1, props2 } = this.props
+    if (props1 !== prevProps.props1) {
+        this.updateMyPluginData()
+    }
+}
 ```
 
-And use it as a custom element
+This function will be called after the `render()` function. 
+It will also receive the properties sent by MARCEL in the `this.props` object, but it also has the previous set of properties in the argument `prevProps`.
+This can be useful to check if your properties have changed, and only trigger expensive API call if the properties did changed.
 
-```html
-<datetime-item></datetime-item>
+Finally, you should create an instance of your class:
+```javascript
+const instance = new MyPlugin()
 ```
 
-## Build
+Additionally, you can add the line:
+```javascript
+Marcel.Debug.changeProps({props1: "new value"})
+```
+
+At the end of your file to force MARCEL to call `render()` and `propsDidChange(prevProps)` when you load the page which is really useful when you are still developing and debugging the plugin.
+
+### Trying out your plugin
 
-The best way to build your plugin is to use macel-builder.
+Since your plugin is just a normal website, you should be able to test it by just opening the index.html in your browser.
+You can also use a development web server like `serve` to autoreload your page while you are working on it.
```

Comment:
You can specify here that you can use `Marce.Debug.changeProps` in the dev console to simulate props changement comming from MARCEL.